libs/packages/components/src/lib/autocomplete-search/autocomplete-search.component.ts

```diff
@@ -258,8 +256,15 @@ export class SDSAutocompleteSearchComponent implements ControlValueAccessor {
         this.getResults(this.inputValue || '');
       }
       this.onTouchedCallback();
+      if (this.isAutocompleteWithinModal()) {
+        this.addListener();
+      }
+
     }
   }
+  isAutocompleteWithinModal() {
+    return document.getElementsByClassName('sds-dialog-content').length > 0
```

Comment:
This returns true if there is any sds-dialog-content on current page document no? If we have an autocomplete in page not in modal, and another element opens a modal without autocomplete in it, this method would return true because there is a `sds-dialog-content` element on DOM

Reply:
The sds-dialog-content class will be there only if the content is inside the modal, if the modal open even though the modal has no autocomplete the the background control will not work